### ISSUE TITLE
DL4006 is triggered when using non-bash/ash shells

### ISSUE THREAD

**Issue by onwsk8r** [hadolint/hadolint#977]
- [x] This is a bug report
- [ ] This is a feature request
- [ ] I searched existing issues before opening this one

### Expected behavior

A valid Containerfile lints successfully.

### Actual behavior

I get a big fat red exclamation point in my CI.

### Steps to reproduce the behavior

**Output of `hadolint --version` or
  `docker run --rm hadolint/hadolint hadolint --version` or
  `docker run --rm ghcr.io/hadolint/hadolint hadolint --version`:**

```bash
❯ podman run -t -v ../auspex:/foo -w /foo --rm docker.io/hadolint/hadolint hadolint --version
Haskell Dockerfile Linter 2.12.1
❯ podman run -t -v ../auspex:/foo -w /foo --rm docker.io/hadolint/hadolint hadolint Dockerfile
Dockerfile:4 DL4006 warning: Set the SHELL option -o pipefail before RUN with a pipe in it. If you are using /bin/sh in an alpine image or if your shell is symlinked to busybox then consider explicitly setting your SHELL to /bin/ash, or disable this check
```

**Dockerfile** (if relevant)

```Dockerfile
FROM docker.io/tensorflow/tensorflow:latest-gpu
WORKDIR /usr/local/src
SHELL ["/bin/sh", "-o", "pipefail", "-c"]
RUN curl -fsSL http://example.com/foo.tar.gz | tar zx
```

**Additional environment details (OS, stack version, etc.)**

This Dockerfile lints fine:

```Dockerfile
FROM docker.io/tensorflow/tensorflow:latest-gpu
WORKDIR /usr/local/src
SHELL ["/bin/bash", "-o", "pipefail", "-c"]
RUN curl -fsSL http://example.com/foo.tar.gz | tar zx
```

```shell
$ podman run -t -v ../auspex:/foo -w /foo --rm docker.io/hadolint/hadolint hadolint Dockerfile
$
```

See? Nothing!

**Comment by maxschae4** [comment]
I bumped into this when using `/usr/bin/bash`, which was confusing since `/bin` is merely a symlink to `/usr/bin` in my particular case.

It turns out [rule DL4006](https://github.com/hadolint/hadolint/blob/master/src/Hadolint/Rule/DL4006.hs#L36) is extremely naive, requiring an exact string match of a relatively small set of possibilities:

```
validShell <- ["/bin/bash", "/bin/zsh", "/bin/ash", "bash", "zsh", "ash"]
```
